crates/holochain/src/core/workflow/publish_dht_ops_workflow/publish_query.rs

```diff
@@ -60,6 +59,29 @@ where
             (DhtOp.last_publish_time IS NULL OR DhtOp.last_publish_time <= :recency_threshold)
             AND
             DhtOp.receipts_complete IS NULL
+
+            UNION
+            ALL
+
+            SELECT
+            Warrant.blob as action_blob,
+            Warrant.author as author,
+            LENGTH(Warrant.blob) AS action_size,
+            0 AS entry_size,
+            NULL as entry_blob,
+            DhtOp.type as dht_type,
+            DhtOp.hash as dht_hash
+            FROM Warrant
+            JOIN
+            DhtOp ON DhtOp.action_hash = Warrant.hash
+            WHERE
+            Warrant.author = :author
+            AND
+            DhtOp.withhold_publish IS NULL
+            AND
+            (DhtOp.last_publish_time IS NULL OR DhtOp.last_publish_time <= :recency_threshold)
+            AND
+            DhtOp.receipts_complete IS NULL
```

Comment:
We can't rely on this to stop warrants being published, validation isn't expected to return a receipt for warrants